src/tests/HDFFile.cxx

```diff
@@ -118,155 +141,323 @@ class T_CommandHandler : public testing::Test {
+      LOG(4, "unit_test.n_msgs_per_batch: {}", x.v);
+      n_msgs_per_batch = x.v;
+    }
+
+    vector<SourceDataGen> sources;
+    for (int i1 = 0; i1 < n_sources; ++i1) {
+      sources.emplace_back();
+      auto &s = sources.back();
+      // Currently, we assume only one topic!
+      s.topic = "topic.with.multiple.sources";
+      s.source = fmt::format("for_example_motor_{:04}", i1);
+      s.pregenerate(n_msgs_per_source, n_events_per_message);
+    }
+
+    for (auto &source : sources) {
+      // LOG(4, "msgs: {}  {}", source.source, source.msgs.size());
+    }
+
+    rapidjson::Document json_command;
+    {
+      using namespace rapidjson;
+      auto &j = json_command;
+      auto &a = j.GetAllocator();
+      j.SetObject();
+      Value nexus_structure;
+      nexus_structure.SetObject();
+      Value streams;
+      streams.SetArray();
+      for (auto &source : sources) {
+        Value v;
+        v.SetObject();
+        nexus_structure.AddMember(StringRef(source.source.c_str()), v, a);
+        Value v2;
+        v2.SetObject();
+        v2.AddMember("topic", StringRef(source.topic.c_str()), a);
+        v2.AddMember("source", StringRef(source.source.c_str()), a);
+        v2.AddMember("nexus_path",
+                     Value(fmt::format("/{}", source.source).c_str(), a), a);
+        streams.PushBack(v2, a);
+      }
+      j.AddMember("nexus_structure", nexus_structure, a);
+      j.AddMember("streams", streams, a);
+      {
+        Value v;
+        v.SetObject();
+        v.AddMember("file_name", StringRef("tmp-ev42.h5"), a);
+        j.AddMember("file_attributes", v, a);
       }
-      fwt->demuxers().at(0).process_message(msg.data, msg.size);
+      j.AddMember("cmd", StringRef("FileWriter_new"), a);
     }
-    auto t2 = CLK::now();
-    LOG(7, "processing done in {}",
-        std::chrono::duration_cast<MS>(t2 - t1).count());
-    LOG(7, "finishing...");
-    fwt.reset();
-    LOG(7, "done.");
 
-    herr_t err;
+    auto cmd = json_to_string(json_command);
+    // LOG(4, "command: {}", cmd);
 
-    auto fid = H5Fopen(string(fname).c_str(), H5F_ACC_RDONLY, H5P_DEFAULT);
-    ASSERT_GE(fid, 0);
+    auto &d = json_command;
+    auto fname = get_string(&d, "file_attributes.file_name");
+    ASSERT_GT(fname.v.size(), 8);
 
-    auto g0 = H5Gopen(fid, "/entry-01/instrument-01/events-01", H5P_DEFAULT);
-    ASSERT_GE(g0, 0);
+    FileWriter::CommandHandler ch(main_opt, nullptr);
 
-    auto ds = H5Dopen2(fid, "/entry-01/instrument-01/events-01/event_id",
-                       H5P_DEFAULT);
-    ASSERT_GE(ds, 0);
+    using DT = uint32_t;
+    int const feed_msgs_times = 1;
+    int const seed = 2;
+    std::mt19937 rnd_nn;
 
-    auto dt = H5Dget_type(ds);
-    ASSERT_GE(dt, 0);
-    ASSERT_EQ(H5Tget_size(dt), sizeof(DT));
+    for (int file_i = 0; file_i < 1; ++file_i) {
+      unlink(string(fname).c_str());
+
+      Msg msg;
+      msg.data = (char *)cmd.data();
+      msg.size = cmd.size();
```

Comment:
I think this is preferable to having the uninitialised type and then populating in parts:
```
Msg msg = {
    (char *)cmd.data(),
    cmd.size(),
};
```